src/app/Rules/CreateRule.tsx

```diff
@@ -170,11 +173,25 @@ const Comp = () => {
 
   // FIXME we query ourselves to populate the list of templates, since Rules can apply to any target. Is this better than making the user write the event specifier manually,
   // or at least make them write the name manually and choose TARGET/CUSTOM from a dropdown?
+
   React.useEffect(() => {
     addSubscription(
-      context.api.doGet<EventTemplate[]>(`targets/localhost:0/templates`).subscribe(setTemplates)
+      context.target.target()
+      .pipe(
+        concatMap(target => 
+          iif(
```

Comment:
Nice use of `iif` here, very clever.

If the user has not selected any target (ie it's null/undefined/`NO_TARGET`), do we need to show anything at all in the template selection dropdown? I think we could just leave the dropdown empty instead.

Reply:
Right, I was gonna ask about that. If we do that, then we force the user to use the `Match Expression Evaluator` to select a target to create a rule. 

> Now, new question: what templates do we display when the user has not selected any target? Perhaps they do not currently have any target to select, but they want to create an Automated Rule in advance before deploying their application.
> 
> In that case perhaps we should query ourselves and filter out the one with `Provider: Cryostat`

Going off what you said in the related issue, it seems this would entail different implementation and functionality than what you said before, unless I'm misinterpreting.

Reply:
Ah fair, good point. Okay, what you have now makes sense.